framework/src/actions/CreateProblemDefaultAction.C

```diff
@@ -0,0 +1,95 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CreateProblemDefaultAction.h"
+
+#include "Factory.h"
+#include "FEProblem.h"
+#include "EigenProblem.h"
+#include "MooseApp.h"
+#include "CreateExecutionerAction.h"
+#include "CreateProblemAction.h"
+
+registerMooseAction("MooseApp", CreateProblemDefaultAction, "create_problem_default");
+registerMooseAction("MooseApp", CreateProblemDefaultAction, "determine_system_type");
+
+template <>
+InputParameters
+validParams<CreateProblemDefaultAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addPrivateParam<bool>("_solve");
+  return params;
+}
+
+CreateProblemDefaultAction::CreateProblemDefaultAction(InputParameters parameters)
+  : Action(parameters)
+{
+}
+
+void
+CreateProblemDefaultAction::act()
+{
+  if (_current_task == "determine_system_type")
+  {
+    /**
+     * Determine whether the Executioner is derived from EigenExecutionerBase and
+     * set a flag on MooseApp that can be used during problem construction.
+     */
```

Comment:
It is useless to use doxygen comments inside code. Simple `//` is fine.

Reply:
Right, copy-paste...